app/js/components/manageApps/ManageModules.jsx

```diff
@@ -0,0 +1,158 @@
+import React, { Component } from 'react';
+import axios from 'axios';
+
+class ManageModules extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      installedModules: [],
+      action: null,
+      stopping: false,
+      starting: false,
+    };
+    this.handleAction = this.handleAction.bind(this);
+    this.actionRunning = this.actionRunning.bind(this);
+  }
+
+  componentWillMount() {
```

Comment:
`componentDidMount` is considered the best life cycle method to handle ajax request. If the ajax request fails and you are performing your asynchronous action in the `componentWillMount` lifecycle method, the page wouldn't render at all, resulting in a blank screen. Not good for user experience.

Reply:
👍